### PR TITLE
LPS-119068 Broken URL when going back from Layout Reports settings

### PR DESCRIPTION
**Pull request by 4lejandrito**
https://issues.liferay.com/browse/LPS-130086

**Steps to reproduce:**

* Go to http://localhost:8080/
* Click on the Page Audit icon
* Click on "Connect to Google PageSpeed"
* Click on the Back button or Cancel button in the settings page to go back to the previous URL

### CODE REVIEW

**Comment by 4lejandrito** [comment]
ci:test:sf

**Comment by 4lejandrito** [comment]
ci:test:relevant

**Comment by 4lejandrito** [comment]
ci:test:sf

**Comment by 4lejandrito** [comment]
ci:test:relevant

**Review by cgoncas — APPROVED** [review]
Awesome @4lejandrito !!!!

**Comment by 4lejandrito** [comment]
Not so awesome. The issue still happens intermittently :(

**Comment by dgarciasarai** [comment]
FYI @4lejandrito, one little thing to take into account, we use the parent LPS ticket for the commits, you can use sub-tasks to control or divide the work, or you may have some impedibugs, but we use the parent LPS in case some day a backport is needed to not miss any commit. 

Let's keep investigating and working on this bug to found a fix. Thanks! 👍

**Comment by 4lejandrito** [comment]
I'll have to send new commits. I'll update the LPS on the next push. Thanks!

File: modules/apps/layout/layout-reports-web/src/main/java/com/liferay/layout/reports/web/internal/display/context/LayoutReportsDisplayContext.java [hunk 1]
```diff
@@ -80,7 +80,7 @@ public LayoutReportsDisplayContext(
 			WebKeys.THEME_DISPLAY);
 	}
 
-	public Map<String, Object> getData() {
+	public Map<String, Object> getData() throws PortalException {
```
Comment (cgoncas) [inline modules/apps/layout/layout-reports-web/src/main/java/com/liferay/layout/reports/web/internal/display/context/LayoutReportsDisplayContext.java:83]: Better if we don't trow the exception, I mean, if the back URL is broken is OK, we can handle it, we should not stop rendering the React component.

File: modules/apps/layout/layout-reports-web/src/main/java/com/liferay/layout/reports/web/internal/display/context/LayoutReportsDisplayContext.java [hunk 2]
```diff
@@ -211,9 +212,7 @@ private String _getConfigureGooglePageSpeedURL(
 				"mvcRenderCommandName",
 				"/configuration_admin/edit_configuration");
 			portletURL.setParameter(
-				"redirect",
-				_portal.getCurrentCompleteURL(
-					_portal.getHttpServletRequest(portletRequest)));
+				"redirect", _portal.getLayoutURL(_themeDisplay));
```
Comment (cgoncas) [inline modules/apps/layout/layout-reports-web/src/main/java/com/liferay/layout/reports/web/internal/display/context/LayoutReportsDisplayContext.java:215]: That change makes totally sense, I need to use it in oder to calculate a valid canonical URL, very good catch :)

File: modules/apps/layout/layout-reports-web/src/main/java/com/liferay/layout/reports/web/internal/display/context/LayoutReportsDisplayContext.java (hunk 1 above)
Comment (cgoncas) [reply modules/apps/layout/layout-reports-web/src/main/java/com/liferay/layout/reports/web/internal/display/context/LayoutReportsDisplayContext.java:83]: I think we have a method called _getCompleteURL, that call this method and if yoiu find an exception returns the currentCompleteURL, what do you think about using it?

File: modules/apps/layout/layout-reports-web/src/main/java/com/liferay/layout/reports/web/internal/display/context/LayoutReportsDisplayContext.java (hunk 2 above)
Comment (cgoncas) [reply modules/apps/layout/layout-reports-web/src/main/java/com/liferay/layout/reports/web/internal/display/context/LayoutReportsDisplayContext.java:215]: I think we have a method called _getCompleteURL, that call this method and if yoiu find an exception returns the currentCompleteURL, what do you think about using it?

**Comment by 4lejandrito** [comment]
Hi @cgoncas,

We've decided to go for a different fix (I just push forced). See https://github.com/liferay-tango/liferay-portal/pull/1056/commits/c68b2caa344ef82036a66f703612e02a2dea3e09.

The problem with the other fix is that we don't go back to the exact same URL, we just go back to the root URL of the page.

What do you think?

File: modules/apps/layout/layout-reports-web/src/main/java/com/liferay/layout/reports/web/internal/display/context/LayoutReportsDisplayContext.java (hunk 1 above)
Comment (4lejandrito) [reply modules/apps/layout/layout-reports-web/src/main/java/com/liferay/layout/reports/web/internal/display/context/LayoutReportsDisplayContext.java]: Valid point, but see https://github.com/liferay-tango/liferay-portal/pull/1056#issuecomment-815031181. If for some reason we rollback to this previous solution I will do this.

**Comment by dgarciasarai** [comment]
ci:test:sf

**Comment by dgarciasarai** [comment]
ci:test:relevant

**Comment by 4lejandrito** [comment]
ci:test:sf

**Comment by 4lejandrito** [comment]
ci:test:relevant

**Comment by 4lejandrito** [comment]
ci:test:sf

**Comment by 4lejandrito** [comment]
ci:test:relevant

**Comment by 4lejandrito** [comment]
ci:forward

**Comment by 4lejandrito** [comment]
ci:stop

**Comment by 4lejandrito** [comment]
ci:forward

**Comment by cgoncas** [comment]
@4lejandrito !!!

This is the most beautiful detailed commit message I have ever see.

Thanks!!

**Comment by 4lejandrito** [comment]
ci:forward

**Pull request closed without merging**